src/context.rs

```diff
@@ -63,11 +65,8 @@ impl<'a> QueryMatchContext<'a> {
             .push(violation);
     }
 
-    pub fn get_node_text(&self, node: Node) -> &str {
-        match &self.file_contents {
-            RopeOrSlice::Slice(file_contents) => node.utf8_text(file_contents).unwrap(),
-            RopeOrSlice::Rope(_) => unimplemented!(),
-        }
+    pub fn get_node_text(&self, node: Node) -> &'a str {
+        get_node_text(node, self.file_contents)
```

Comment:
Extracted this to be testable without needing to construct `QueryMatchContext` instances (really to be able to test `get_tokens()`, which relies on this)

There's sort of a parallel where ESLint exposes a `SourceCode` sub-object on the `context` and it handles things like this (and probably isn't aware of things like rules)

I'm not sure I like the ergonomics of that though, I think I like "directly asking the `context`" for things